Max9545/What-s-Cookin-Refactory
# Gaj/feature/search bar

## *Purpose:*
*What does this merge do? Check all that apply.*


## *Description: What does this PR do?*
*Briefly, but clearly describe what you did, and explain any new code.*

This feature manages the functionality of the search bar. If a user is to enter in a specific recipe name, they will be shown that recipe. If they enter in a tag, say starter, the options for starters will be shown. If they list it by a possible ingredient such as banana, they will see those options as well. However, if a user enters in a general word such as chicken or pork, they will be given a list of all recipe possibilities that fulfill that criteria based on the recipe name, ingredients, or tags.  This feature also addresses if the user enters in different capitalization and that the user will only ever see one copy of an recipe in their list of possibilities. 


## *Instructions for reviewer:*
*Where should the reviewer start?*
*How should this be manually tested?*
*Any background context you want to provide?*
*What are the relevant tickets?*

Please begin by testing search bar functionality. 
Run npm start on the local copy of the repo as well as npm start on the API associated with the data. 
Then, please check the search bar by entering different values. 

Significant changes were made inside of the Cookbook class. All logic begins there. Testing was altered as well. Lines 46 - 56 were attempts at SAD path testing. Run `npm test test/cookbook-test.js` to confirm all tests are still passing. 

Additional logic was added to the final function in scripts that would allow for the dom to be updated in case the search resulted in nothing (scripts line 154).  This functionality can be viewed on  domUpdates line 46-48. 

Data files were moved from `src` to `test`. 

Tickets closed: - #54 - #48 - #3 - #20 - #34

## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*

Additional styling should be added to the no-results element. 

## Checklist:
